cocos/renderer/gfx-gles3/GLES3Device.cpp

```diff
@@ -282,6 +257,221 @@ void GLES3Device::bindContext(bool bound) {
     _gpuContext->bindContext(bound);
 }
 
+void GLES3Device::initFormatFeature() {
+    _textureExclusive.fill(true);
+
+    const FormatFeature completeFeature = FormatFeature::RENDER_TARGET | FormatFeature::SAMPLED_TEXTURE | FormatFeature::LINEAR_FILTER | FormatFeature::STORAGE_TEXTURE;
+    FormatFeature       tempFeature     = completeFeature;
+
+    // builtin
+    tempFeature = completeFeature | FormatFeature::VERTEX_ATTRIBUTE;
+
+    _formatFeatures[toNumber(Format::R8)]    = tempFeature;
+    _formatFeatures[toNumber(Format::RG8)]   = tempFeature;
+    _formatFeatures[toNumber(Format::RGB8)]  = tempFeature;
+    _formatFeatures[toNumber(Format::RGBA8)] = tempFeature;
+
+    tempFeature = completeFeature;
+
+    _formatFeatures[toNumber(Format::R8SN)]    = tempFeature;
+    _formatFeatures[toNumber(Format::RG8SN)]   = tempFeature;
+    _formatFeatures[toNumber(Format::RGB8SN)]  = tempFeature;
+    _formatFeatures[toNumber(Format::RGBA8SN)] = tempFeature;
+    _formatFeatures[toNumber(Format::R5G6B5)]  = tempFeature;
+    _formatFeatures[toNumber(Format::RGBA4)]   = tempFeature;
+    _formatFeatures[toNumber(Format::RGB5A1)]  = tempFeature;
+    _formatFeatures[toNumber(Format::RGB10A2)] = tempFeature;
+
+    _formatFeatures[toNumber(Format::SRGB8)]    = tempFeature;
+    _formatFeatures[toNumber(Format::SRGB8_A8)] = tempFeature;
+
+    _formatFeatures[toNumber(Format::R11G11B10F)] = tempFeature;
+    _formatFeatures[toNumber(Format::RGB9E5)]     = tempFeature;
+
+    _formatFeatures[toNumber(Format::DEPTH)]         = tempFeature;
+    _formatFeatures[toNumber(Format::DEPTH_STENCIL)] = tempFeature;
+
+    removeFlags(tempFeature, FormatFeature::SAMPLED_TEXTURE);
```

Comment:
Don't do this here, the flags should be declared as straightforward as possible in advance, with each variable names reflecting the accurate semantics:
```cpp
    constexpr FormatFeature textureFeaturesNoFiltering = FormatFeature::RENDER_TARGET | FormatFeature::SAMPLED_TEXTURE | FormatFeature::STORAGE_TEXTURE;
    constexpr FormatFeature allTextureFeatures = textureFeaturesNoFiltering | FormatFeature::LINEAR_FILTER;
    constexpr FormatFeature allFeatures = allTextureFeatures  | FormatFeature::VERTEX_ATTRIBUTE;
```
All the procedural bitwise manipulations in the middle of format declarations will only make the code hard to follow or modify.